### PR TITLE
Run `release` command from Procfile in Deploy flow

### PR DESCRIPTION
**Pull request by drgarcia1986**
This is the most wanted feature according to our users!

This feature enables devs to run tasks (e.g. migrations, APM deploy notify, etc) before update the current deploy.

The release command will be run between build phase and deploy update phase. If some error comes across in release phase the deployed content won't be updated.

---
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa-api/226)

### CODE REVIEW

File: pkg/server/deploy/deploy.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ func (ops *DeployOperations) Deploy(user *storage.User, appName string, tarBall
 		slugURL := fmt.Sprintf("%s/slug.tgz", buildDest)
 		releaseCmd := confFiles.Procfile[ProcfileReleaseCmd]
 		if confFiles.Procfile != nil && releaseCmd != "" {
-			if err := ops.runReleaseCmd(a, slugURL, w); err != nil {
+			if err := ops.runReleaseCmd(a, deployId, slugURL, w); err != nil {
 				log.WithError(err).Errorf("Running release command %s in app %s", releaseCmd, appName)
```
Comment (aguerra) [inline pkg/server/deploy/deploy.go]: Maybe log deployId on errors, don't know if it's useful for debugging.

File: pkg/server/deploy/errors.go [hunk 2]
```diff
@@ -7,5 +7,6 @@ import (
 
 var (
 	ErrBuildFail             = status.Errorf(codes.Unknown, "Build return a non zero value")
+	ErrReleaseFail           = status.Errorf(codes.Unknown, "Release command return a non zero value")
 	ErrInvalidTeresaYamlFile = status.Errorf(codes.InvalidArgument, "Invalid Teresa Yaml file")
```
Comment (drgarcia1986) [inline pkg/server/deploy/errors.go]: `y + p` sorry

File: pkg/server/deploy/deploy.go (hunk 1 above)
Comment (drgarcia1986) [inline pkg/server/deploy/deploy.go]: Why not? I'll add

**Comment by drgarcia1986** [comment]
@aguerra updated

**Pull request merged**